examples/hif2a/fit_to_multiple_rbfes.py

```diff
@@ -0,0 +1,431 @@
+        stddev = {np.std(nonbonded_du_dl):.3f}
+    """
+    print(message)
+
+
+def _mean_du_dlambda(result):
+    """summarize result of rfe.host_edge into mean du/dl
+
+    TODO: refactor where this analysis step occurs
+    """
+    bonded_du_dl, nonbonded_du_dl, _ = result
+    return np.mean(bonded_du_dl + nonbonded_du_dl)
+
+
+def _update_combined_handle_and_grads(ghs, combined_handle_and_grads):
+    # use gradient information from the endpoints
+    for (grad_lhs, handle_type_lhs), (grad_rhs, handle_type_rhs) in zip(ghs[0], ghs[-1]):
+        assert handle_type_lhs == handle_type_rhs  # ffs are forked so the return handler isn't same object as that of ff
+        grad = grad_rhs - grad_lhs
+
+        # complex - solvent
+        if handle_type_lhs not in combined_handle_and_grads:
+            combined_handle_and_grads[handle_type_lhs] = grad
+        else:
+            combined_handle_and_grads[handle_type_lhs] -= grad
+
+
+class ThermodynamicIntegrationResult:
+    def __init__(self, lambda_schedule, results):
+        self.lambda_schedule = lambda_schedule
+        self.results = results
+
+    @staticmethod
+    def _save_result(name, lamb, result, rfe, configuration):
+        """
+        TODO: more compact way to store the relevant information about which transformation was computed
+        TODO: include type hint here for ambiguous argument name "result"
+        TODO: add units
+        TODO: move this into part of a .save() method for a RelativeFreeEnergy result object?
+        TODO: use state index i rather than lambda in filename
+        TODO: include also dU/dparams here?
+        """
+        # unpack result tuple
+        bonded_du_dl, nonbonded_du_dl, grads_and_handles = result
+
+        # save du/dlambda trajectories
+        du_dl_filename = path_to_results.joinpath(
+            f"du_dlambda trajectories ({name}, lambda={lamb:.3f})")
+        print(f'saving bonded and nonbonded du/dlambda trajectories to {du_dl_filename}...')
+        np.savez(du_dl_filename, bonded_du_dl=bonded_du_dl, nonbonded_du_dl=nonbonded_du_dl, rfe=rfe,
+                 configuration=configuration)
+
+    def save(self, name=''):
+        for lamb, result in zip(self.lambda_schedule, self.results):
+            self._save_result(name, lamb, result, rfe, configuration)
+
+
+# TODO: shorten this function
+def predict_dG_and_grad(rfe: RelativeFreeEnergy, conf: Configuration, client: AbstractClient):
+    """
+    rfe defines the free energy transformation,
+    conf specifies the computational details
+    client specifies a multi-node distributed computing client environment
+
+    TODO: add an argument for force field parameters, then register this function with Jax as value_and_grad or similar...
+    """
+
+    # build the complex system
+    complex_system, complex_coords, _, _, complex_box, _ = builders.build_protein_system(
+        path_to_protein)
+    # TODO: optimize box
+    complex_box += np.eye(3) * 0.1  # BFGS this later
```

Comment:
We definitely will need to fix this - this will affect the densities for sure. A crazy idea/hack without resorting to BFGS is actually that we can find the offending water molecules (not atoms) and just lower them from 4D like the we do for the ligand.